File: 프로그래머스/unrated/181188. 요격 시스템/요격 시스템.cpp
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

bool cmp(vector<int>& a, vector<int>& b) {
    return a[1] < b[1];
}

int solution(vector<vector<int>> targets) {
    int answer = 0;
    sort(targets.begin(), targets.end(), cmp);
    int last = -1;
    for (auto target : targets) {
        if (target[0] >= last) {
            answer++;
            last = target[1];
        }
    }
    return answer;
}

File: 프로그래머스/2/181188. 요격 시스템/요격 시스템.java
import java.util.*;
import java.util.stream.*;

class Solution {
    public int solution(int[][] targets) {
        Arrays.sort(targets, Comparator.comparingInt((int[] t) -> t[1]));
        int answer = 0, last = -1;
        for (int[] target : targets) {
            if (target[0] > last) {
                answer++;
                last = target[1] - 1;
            }
        }
        return answer;
    }
}